pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -489,7 +489,7 @@ func (b *EndpointBuilder) filterIstioEndpoint(ep *model.IstioEndpoint, svcPort *
 		return false
 	}
 	// If we don't know the address we must eventually use a gateway address
-	if ep.Address == "" && (!b.gateways().IsMultiNetworkEnabled() || b.proxy.InNetwork(ep.Network)) {
+	if len(ep.Addresses) == 0 && (!b.gateways().IsMultiNetworkEnabled() || b.proxy.InNetwork(ep.Network)) {
```

Comment:
Doesn't endpoint_builder.go have a test file? Can you modify it for testing this?

Reply:
Yes, it has, do you mean the test case for `filterIstioEndpoint`?

Reply:
Hi @jacob-delgado & @keithmattix, yes, I added the `TestFilterIstioEndpoint` for this changes.